rotkehlchen/chain/bitcoin/__init__.py

```diff
@@ -9,62 +7,102 @@
+        for entry in btc_resp['addresses']:
+            balances[entry['address']] = satoshis_to_btc(FVal(entry['final_balance']))
+    return balances
+
+
+_BITCOIN_BALANCE_APIS = {
+    'blockchain.info': _query_blockchain_info,
+    'blockstream.info': _query_blockstream_info,
+    'mempool.space': _query_mempool_space,
+}
+
+
+def get_bitcoin_addresses_balances(
+        accounts: list[BTCAddress],
+) -> dict[BTCAddress, FVal]:
+    """Queries bitcoin balance APIs for the balances of accounts
 
     May raise:
-    - RemotError if there is a problem querying blockchain.info or blockstream
+    - RemoteError if there is a problem querying bitcoin balance APIs
     """
-    source = 'blockchain.info'
-    balances: Dict[BTCAddress, FVal] = {}
-    try:
-        if _have_bc1_accounts(accounts):
-            # if 1 account is bech32 we have to query blockstream. blockchaininfo won't work
-            source = 'blockstream'
-            balances = {}
-            for account in accounts:
-                url = f'https://blockstream.info/api/address/{account}'
-                response_data = request_get_dict(url=url, handle_429=True, backoff_in_seconds=4)
-                stats = response_data['chain_stats']
-                balance = int(stats['funded_txo_sum']) - int(stats['spent_txo_sum'])
-                balances[account] = satoshis_to_btc(balance)
-        else:
-            # split the list of accounts into sublists of 80 addresses per list to overcome:
-            # https://github.com/rotki/rotki/issues/3037
-            accounts_chunks = [accounts[x:x + 80] for x in range(0, len(accounts), 80)]
-            for accounts_chunk in accounts_chunks:
-                params = '|'.join(accounts_chunk)
-                btc_resp = request_get_dict(
-                    url=f'https://blockchain.info/multiaddr?active={params}',
-                    handle_429=True,
-                    # If we get a 429 then their docs suggest 10 seconds
-                    # https://blockchain.info/q
-                    backoff_in_seconds=10,
-                )
-                for entry in btc_resp['addresses']:
-                    balances[entry['address']] = satoshis_to_btc(FVal(entry['final_balance']))
-    except (
+    errors: dict[str, Exception] = {}
+    for name, api in _BITCOIN_BALANCE_APIS.items():
+        try:
+            balances = api(accounts)
+        except (
             requests.exceptions.RequestException,
             UnableToDecryptRemoteData,
             requests.exceptions.Timeout,
-    ) as e:
-        raise RemoteError(f'bitcoin external API request for balances failed due to {str(e)}') from e  # noqa: E501
-    except KeyError as e:
-        raise RemoteError(
-            f'Malformed response when querying bitcoin blockchain via {source}.'
-            f'Did not find key {e}',
-        ) from e
+            BchNotSupported,
+            RemoteError,
+            KeyError,
+        ) as e:
+            errors[name] = e
```

Comment:
This is not good enough.

1. you would need to stringify the exception. (I think you may do it later)
2. For key error the string is simply the name of the missing attribute. So user does not see it's a key error. Check how we handle KeyErrors in the rest of the code.